File: SjoslagetBackend/Models/BookingPaxItem.cs
using System;

namespace Accidis.Sjoslaget.WebService.Models
{
	public sealed class BookingPaxItem
	{
		string _nationality;

		public Guid Id { get; set; }
		public Guid CabinTypeId { get; set; }
		public string Reference { get; set; }
		public string Group { get; set; }
		public string FirstName { get; set; }
		public string LastName { get; set; }
		public string Gender { get; set; }
		public string Dob { get; set; }

		public string Nationality
		{
			get { return _nationality; }
			set { _nationality = value?.ToUpperInvariant(); }
		}

		public int Years { get; set; }
	}
}

File: SjoslagetFrontend/lib/model/booking_pax_item.dart
import 'json_field.dart';

class BookingPaxItem {
  String cabinType;
  final String cabinTypeId;
  final String reference;
  final String subCruise;
  String group;
  final String firstName;
  final String lastName;
  final String gender;
  final String dob;
  final String nationality;
  final int years;

  BookingPaxItem(this.cabinTypeId, this.reference, this.subCruise, this.group, this.firstName, this.lastName, this.gender, this.dob,
      this.nationality, this.years);

  factory BookingPaxItem.fromMap(Map<String, dynamic> json) => BookingPaxItem(json[CABIN_TYPE_ID], json[REFERENCE], json[SUBCRUISE],
      json[GROUP], json[FIRSTNAME], json[LASTNAME], json[GENDER], json[DOB], json[NATIONALITY], json[YEARS]);

  String get name => '$firstName $lastName';
}
